client/src/app/pages/search/components/SearchMenu.tsx

```diff
@@ -0,0 +1,314 @@
+
+  const [searchValue, setSearchValue] = React.useState<string | undefined>(
+    defaultValue
+  );
+  const [autocompleteOptions, setAutocompleteOptions] = React.useState<
+    React.JSX.Element[]
+  >([]);
+  const [isAutocompleteOpen, setIsAutocompleteOpen] =
+    React.useState<boolean>(false);
+
+  const searchInputRef: React.RefObject<HTMLInputElement> = React.useRef(null);
+  const autocompleteRef: React.RefObject<HTMLInputElement> = React.useRef(null);
+
+  const onChangeSearchValue = (newValue: string) => {
+    if (
+      newValue !== "" &&
+      searchInputRef &&
+      searchInputRef.current &&
+      searchInputRef.current.contains(document.activeElement)
+    ) {
+      setIsAutocompleteOpen(true);
+
+      // When the value of the search input changes, build a list of no more than 10 autocomplete options.
+      // Options which start with the search input value are listed first, followed by options which contain
+      // the search input value.
+      let options: React.JSX.Element[] = entityList
+        .filter(
+          (option) =>
+            option.id.toLowerCase().startsWith(newValue.toLowerCase()) ||
+            option.title?.toLowerCase().startsWith(newValue.toLowerCase()) ||
+            option.description?.toLowerCase().startsWith(newValue.toLowerCase())
+        )
+        .map((option) => (
+          <MenuItem
+            itemId={option.id}
+            key={option.id}
+            description={option.description}
+            to={option.navLink}
+          >
+            {option.title} <Label color={option.typeColor}>{option.type}</Label>
+          </MenuItem>
+        ));
+
+      if (options.length > 10) {
+        options = options.slice(0, 10);
+      } else {
+        options = [
+          ...options,
+          ...entityList
+            .filter(
+              (option: IEntity) =>
+                !option.id.startsWith(newValue.toLowerCase()) &&
+                option.id.includes(newValue.toLowerCase())
+            )
+            .map((option: IEntity) => (
+              <MenuItem itemId={option.id} key={option.id}>
+                {option.id}
+              </MenuItem>
+            )),
+        ].slice(0, 10);
+      }
+
+      // The menu is hidden if there are no options
+      setIsAutocompleteOpen(options.length > 0);
+      setAutocompleteOptions(options);
+    } else {
+      setIsAutocompleteOpen(false);
+    }
+
+    setSearchValue(newValue);
+  };
+
+  const onClearSearchValue = () => {
+    setSearchValue("");
+    onChangeSearch("");
+  };
+
+  const onSubmitInput = () => {
+    onChangeSearch(searchValue);
+  };
+
+  // Whenever an autocomplete option is selected, set the search input value, close the menu, and put the browser
+  // focus back on the search input
+  const onSelect = (e: any, itemId?: string | number | undefined) => {
+    e?.stopPropagation();
+    setSearchValue(itemId as string);
+    setIsAutocompleteOpen(false);
+    searchInputRef.current?.focus();
```

Comment:
It might be just my eyes but I feel like there is a slight delay between the action CLICK and the redirect. Perhaps we could directly call `navigate("/new/url")` here? but I didn't test whether or not that would make it faster.

https://github.com/user-attachments/assets/984fb91b-0ec9-409c-8e53-633d394817ae

Reply:
Yes I see it as well, and I did notice at the time but forgot to go back to it! I'll try it with `navigate`

Reply:
I couldn't fix this, even with `useNavigate`, so I'll be creating an issue to track it